containers/runtime/xds/README.md

```diff
@@ -0,0 +1,124 @@
+
+Another piece of information communicated from the Driver's ready container is
+the type of the test. For "proxied" test, the xds server will remove all
+api_listneners and only serves the socket listener for validation purpose.
+
+After filling in the actual backend services, the xDS server starts listening on
+request and server the configuration created through above steps.
+
+## Options for starting the xDS server
+
+The following example start a xDS server with the initial endpoint update server
+listening on port 18005, serving resource snapshot created based on
+`config/default_config.json`. The server target string passed to gRPC proxyless
+client needs to match the name of the listener resource that handling the
+requests, the listener name is primarily supplied through the
+`default_config.json` file, here the flag is used to validate that at lease one
+of the listener has the name of the server target string.
+
+```shell
+go run main.go
+   -default-config-path config/default_config.json \
+   -endpoint-update-port 18005    \
+   -psm-target-string default_testGrpcListenerName
+
+```
+
+The binary main.go takes the following options:
+
+- -xds-server-port
+
+  The server port of the xds server listening on. This port matches the server
+  port in the field `server_uri` in `../gRPC_bootstrap/bootstrap.json` file and
+  `port_value` in the `static_resources` section in `../envoy/envoy.yaml` file.
+  This filed has the defalt value of `18000`.
+
+- -endpoint-update-port
+
+  The port that endpoint update server is listening on. The default value of the
+  port value is `18005`.
+
+- -node-ID
+
+  The node ID that this snapshot of the resource can be served to. The nodeID
+  has to match `id` field in `node` section in
+  `../gRPC_bootstrap/bootstrap.json` file and `id` field in `node` section in
+  `../envoy/envoy.yaml` file. The default value of this field is `test_id`.
+
+- -default-config-path
+
+  The path of default configuration JSON file that the resource snapshot based
+  on. The default value of the field is `config/default_config.json`.
+
+- -custom-config-path
+
+  The path of the user supplied configuration JSON file. No default value for
+  this field, for more information check section:
+  [Custom configuration of xDS server](#Custom-configuration-of-xDS-server)
+
+- -non-proxied-target-string
+
+  This field only for validation. The listener name that serves non-proxied
+  client. The listener name has to match the server-target-string in
+  xds:///server-target-string, which is passed as target for test client. The
+  source of the truth for name of the listener's names are from the
+  configuration files, the flag here is to make sure that at lease one of the
+  listener resource has the required name.
+
+- -sidecar-listener-port
+
+  This field in only for validation. The listener port that sidecar proxy is
+  listening on. The traffic wish to go through envoy should be send to this
+  port.
+
+- -validation-only
```

Comment:
Suggestion: Maybe call this `-validate-only` instead? Please also add a sentence clarifying that the server will not actually run if this flag is specified. That wasn't clear to me until I read the code.